lib/kafka/consumer.rb

```diff
@@ -340,12 +340,12 @@ def fetch_batches(min_bytes:, max_wait_time:)
             @logger.warn "Partition #{topic}/#{partition} is currently paused, skipping"
           else
             @logger.debug "Fetching batch from #{topic}/#{partition} starting at offset #{offset}"
-            operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
+            @operation.fetch_from_partition(topic, partition, offset: offset, max_bytes: max_bytes)
           end
         end
       end
 
-      operation.execute
+      @operation.execute
```

Comment:
Hmm... couldn't the state live on the individual connections rather than the operation? Calling `fetch_batches` would register a request for a specific correlation id on each connection. Blocking on a future would mean waiting until there's a response with that correlation id on the connection. We could make some of the Connection methods public in order to facilitate this.